app/src/main/java/com/marcossalto/targetmvd/ui/base/BaseFragment.kt

```diff
@@ -0,0 +1,43 @@
+package com.marcossalto.targetmvd.ui.base
+
+import android.app.AlertDialog
+import androidx.fragment.app.Fragment
+import com.marcossalto.targetmvd.R
+import com.marcossalto.targetmvd.ui.custom.LoadingDialog
+
+open class BaseFragment : Fragment(), BaseView {
+
+    private var loadingDialog: LoadingDialog? = null
+
+    override fun showProgress() {
+        if (loadingDialog == null) {
+            loadingDialog = LoadingDialog(requireContext(), null)
+        }
+
+        loadingDialog!!.show()
+    }
+
+    override fun hideProgress() {
+        if (loadingDialog != null) {
+            loadingDialog!!.dismiss()
+        }
+    }
+
+    override fun showError(message: String?) {
+        val builder = AlertDialog.Builder(requireContext())
+        builder.setTitle(getString(R.string.error))
+
+        when (message) {
+            "" -> builder.setMessage(getString(R.string.generic_error))
+            null -> builder.setMessage(getString(R.string.generic_error))
+            else -> builder.setMessage(message)
+        }
+
+        builder.setPositiveButton(getString(R.string.ok)) { dialog, _ ->
+            dialog.cancel()
+        }
```

Comment:
try to use a scope function with the builders
I'd use something like this to assign the message.
```
val builderMessage = if(message.isNullOrBlank()) 
      getString(R.string.generic_error) 
  else 
      message
```

Reply:
Thanks for your suggestion